src/core/reactor_backend.cc

```diff
@@ -1531,31 +1640,89 @@ class reactor_backend_uring final : public reactor_backend {
                 sizeof(iovec) == sizeof(net::fragment)
                 , "net::fragment and iovec should be equivalent");
 
-            iovec* iov = reinterpret_cast<iovec*>(p.fragment_array());
-            msghdr mh = {};
-            mh.msg_iov = iov;
+            ::msghdr mh = {};
+            mh.msg_iov = reinterpret_cast<iovec*>(p.fragment_array());
             mh.msg_iovlen = std::min<size_t>(p.nr_frags(), IOV_MAX);
-            auto r = fd.fd.sendmsg(&mh, MSG_NOSIGNAL);
-            if (!r) {
-                return write_some(fd, p);
+            auto r = fd.fd.sendmsg(&mh, MSG_NOSIGNAL | MSG_DONTWAIT);
+            if (r) {
+                if (size_t(*r) == p.len()) {
+                    fd.speculate_epoll(EPOLLOUT);
```

Comment:
This can throw. Please check that the callers catch the exception and return a future with an exception (for the other calls too).

Reply:
By "this" I meant the call to sendmsg().

Reply:
yeah, `fd.fd.sendmsg()` throws. but if we check the diff in this commit (https://github.com/scylladb/seastar/pull/1235/commits/1a9baa61eeb1358f2183eef2407ad5032e313aaf#diff-33e3c9bf9758fbe5445e34b395c6f07d6b1ea890ba224e605a7e7860f3b5064aR1646) , i made no change in this area. the existing callers already (or should already) catch the expected exceptions. 

> return a future with an exception (for the other calls too).

IIUC, if a continuation body throws, we already set the state in the corresponding promise with the exception. right? in other words, since the underlying future/promise framework has taken care of this, we don't need to do extra work, right?

Reply:
thanks. i see now. will do something like `futurize_invoke()`.

Reply:
hmm, thought i create a clever wrapper do implement this. but the code flow with early returns makes it difficult. so i am just putting it in the unrolled form. but the upside is that it is easier to read. =)

Reply:
I think it's fine to be unrolled here.